### PR TITLE
Run CI steps in parallel

### PR DESCRIPTION
**Pull request by Rory-Powell**
## Description
- Break up the CI job to run individual steps
   - lint
   - build
   - unit test
   - integration test 
- This allows for an easier re-run of any part of the entire ci process (e.g. a flaky test)
- This brings down the overall CI time by about 7 minutes 

Addresses: 
-  https://linear.app/budibase/issue/BUDI-6622/run-ci-steps-in-parallel

Also see:
- https://github.com/Budibase/budibase-pro/pull/115

### CODE REVIEW

**Review by mike12345567 — APPROVED** [review]
It's beautiful, Send it!

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,4 +1,2 @@
 #!/bin/sh
 . "$(dirname "$0")/_/husky.sh"
-
-yarn run lint
```
Comment (adrinr) [inline .husky/pre-commit:4]: 🎉

File: packages/backend-core/tests/utilities/DBTestConfiguration.ts [hunk 2]
```diff
@@ -12,6 +12,10 @@ class DBTestConfiguration {
     this.tenantId = structures.tenant.id()
   }
 
+  reset() {
```
Comment (adrinr) [inline packages/backend-core/tests/utilities/DBTestConfiguration.ts:15]: Is this reset? Or `useNewDb`?

**Review by adrinr — APPROVED** [review]

File: packages/backend-core/tests/utilities/DBTestConfiguration.ts (hunk 2 above)
Comment (Rory-Powell) [reply packages/backend-core/tests/utilities/DBTestConfiguration.ts:15]: Probably the most correct term would be `newTenant` - I'll update

Comment (Rory-Powell) [reply packages/backend-core/tests/utilities/DBTestConfiguration.ts]: https://github.com/Budibase/budibase/pull/9760/commits/6144416d549c2ac773f5511dcd82b46fc21c4b6a

File: .husky/pre-commit (hunk 1 above)
Comment (Rory-Powell) [reply .husky/pre-commit:4]: 💯

**Pull request merged**